src/Components/SmartComponents/HybridInventory/ImmutableDevicesTab/ImmutableDevices.test.js

```diff
@@ -137,6 +137,30 @@ describe('ImmutableDevices', () => {
     );
   });
 
+  test('should hide group filter from the ImmutableDevices federated module when enforce_edge_group is true', async () => {
+    useSelector.mockReturnValue({ enforceEdgeGroups: true })
```

Comment:
Maybe it's better to add some default `total` value too? So that in the tested component all selectors don't evaluate to undefined except enforceEdgeGroups